closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1053 @@
+                    debug.println("Added restricted security properties, with property: " + jdkPropsName + " value: "
+                            + values);
+                }
+            }
+        }
+
+        // For keyStore and keystore.type, old value not needed, just set the new value.
+        String keyStoreType = properties.keyStoreType;
+        if (!isNullOrBlank(keyStoreType)) {
+            props.setProperty("keystore.type", keyStoreType);
+        }
+        String keyStore = properties.keyStore;
+        if (!isNullOrBlank(keyStore)) {
+            // SSL property "javax.net.ssl.keyStore" set at the JVM level via system properties.
+            System.setProperty("javax.net.ssl.keyStore", keyStore);
+        }
+    }
+
+    /**
+     * Check the restricted security properties
+     */
+    private static void restrictsCheck() {
+        // Check restricts object
+        if (restricts == null) {
+            printStackTraceAndExit("Restricted security property is null.");
+        }
+
+        // Check if the SunsetDate expired.
+        if (isPolicySunset(restricts.descSunsetDate)) {
+            printStackTraceAndExit("Restricted security policy expired.");
+        }
+
+        // Check secure random settings.
+        if (isNullOrBlank(restricts.jdkSecureRandomProvider)
+                || isNullOrBlank(restricts.jdkSecureRandomAlgorithm)) {
+            printStackTraceAndExit("Restricted security mode secure random is null.");
+        }
+    }
+
+    /**
+     * Check if restricted security policy is sunset.
+     *
+     * @param descSunsetDate the sunset date from java.security
+     * @return true if the restricted security policy sunset
+     */
+    private static boolean isPolicySunset(String descSunsetDate) {
+        boolean isSunset = false;
+        try {
+            isSunset = LocalDate.parse(descSunsetDate, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                    .isBefore(LocalDate.now());
+        } catch (DateTimeParseException except) {
+            printStackTraceAndExit(
+                    "Restricted security policy sunset date is incorrect, the correct format is yyyy-MM-dd.");
+        }
+
+        if (debug != null) {
+            debug.println("Restricted security policy is sunset: " + isSunset);
+        }
+        return isSunset;
+    }
+
+    /**
+     * Check if the input string is null or blank.
+     *
+     * @param string the input string
+     * @return true if the input string is null or blank
+     */
+    private static boolean isNullOrBlank(String string) {
+        return (string == null) || string.isBlank();
+    }
+
+    private static void printStackTraceAndExit(Exception exception) {
+        exception.printStackTrace();
+        System.exit(1);
+    }
+
+    private static void printStackTraceAndExit(String message) {
+        printStackTraceAndExit(new RuntimeException(message));
+    }
+
+    /**
+     * This nested class is for save and operate the restricted security
```

Comment:
Again, "nested" isn't relevant to the purpose of this class; and the description doesn't read well.